messaging/src/main/java/org/axonframework/messaging/annotation/HandlerComparator.java

```diff
@@ -26,13 +26,19 @@
  */
 public final class HandlerComparator {
 
-    private static final Comparator<MessageHandlingMember<?>> INSTANCE = Comparator
-            .comparing((Function<MessageHandlingMember<?>, Class<?>>) MessageHandlingMember::payloadType,
-                       HandlerComparator::compareHierarchy)
-            .thenComparing(
-                    Comparator.comparingInt((ToIntFunction<MessageHandlingMember<?>>) MessageHandlingMember::priority)
-                            .reversed())
-            .thenComparing(m -> m.unwrap(Executable.class).map(Executable::toGenericString).orElse(m.toString()));
+    private static final Comparator<MessageHandlingMember<?>> INSTANCE =
```

Comment:
Nit: So, nothing on this class' javadoc describes what the ordering is when the comparator is actually used. Might be nice to include this to provide a quicker glance than having to figure out this `static Comparator INSTANCE` here.